### ISSUE TITLE
how to convert torch.fx model to Tensorrt engine, can Trt parse the onnx model saved by torch.fx directly?

### ISSUE THREAD

**Issue by cqray1990** [NVIDIA/TensorRT#2786]
## Environment

**TensorRT Version**: 
**NVIDIA GPU**: 
**NVIDIA Driver Version**: 
**CUDA Version**: 
**CUDNN Version**: 
**Operating System**: 
**Python Version (if applicable)**: 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**: 

how to convert torch.fx model to Tensorrt engine, how to get  the onnx model with Q/DQ , and then parsed the model by Tensorrt?

**Comment by zerollzeng** [comment]
> how to get the onnx model with Q/DQ

check https://github.com/NVIDIA/TensorRT/tree/release/8.6/tools/pytorch-quantization ?



> how to convert torch.fx model to Tensorrt engine

Can you export it to onnx? if not, then you need to export it as a custom onnx operators, and implement a TRT plugin for it. see https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#extending

**Comment by cqray1990** [comment]
> > how to get the onnx model with Q/DQ
> 
> check https://github.com/NVIDIA/TensorRT/tree/release/8.6/tools/pytorch-quantization ?
> 
> > how to convert torch.fx model to Tensorrt engine
> 
> Can you export it to onnx? if not, then you need to export it as a custom onnx operators, and implement a TRT plugin for it. see https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#extending
i found pytorch-quantization it aim at PTQ ,  But How to export an QAT Model and covert to tensorrt?

**Comment by cqray1990** [comment]
i found pytorch-quantization it aim at PTQ , But How to export an QAT Model and covert to tensorrt?
@pbridger @zerollzeng

**Comment by cqray1990** [comment]
> No, it's for QAT, please check the user guide.

@ @zerollzeng  i read the user guide and the code in tools/pytorch-quantization/
examplses,

 at the begining,there is no model trained,  it is PTQ , it should load a model which is trained by yourself,not an pretain model of resnet50,

  ## Initial accuracy evaluation
    criterion = nn.CrossEntropyLoss()
    with torch.no_grad():
        print('Initial evaluation:')
        top1_initial = evaluate(model, criterion, data_loader_test, device="cuda", print_freq=args.print_freq)

    ## Calibrate the model
    with torch.no_grad():
        calibrate_model(
            model=model,
            model_name=args.model_name,
            data_loader=data_loader_train,
            num_calib_batch=args.num_calib_batch,
            calibrator=args.calibrator,
            hist_percentile=args.percentile,
            out_dir=args.out_dir)

then : the code below is QAT?

  for epoch in range(args.num_finetune_epochs):
        # Training a single epch #train_one_epoch(model, criterion, optimizer, data_loader, device, epoch, args, model_ema, scaler)
        train_one_epoch(model, criterion, optimizer, data_loader_train, device, epoch,args)
        lr_scheduler.step()

**Comment by ttyio** [comment]
@cqray1990 , you need a trained model first, and use QAT to fine-tune your model. For the torch.fx to trt flow, could you check https://pytorch.org/TensorRT/tutorials/getting_started_with_fx_path.html, thanks!

**Comment by ttyio** [comment]
closing since no activity for more than 3 weeks, pls reopen if you still have question, thanks!
